Assets/Scripts/Actors/ActorComponents/ActorPhysics.cs

```diff
@@ -400,10 +347,139 @@ public void JumpMovement( Vector3 inputVec )
 		}
 	}
 
-	protected void ChangeState( ActorStates toState )
+	public void StartGlide()
+	{
+		if ( !_isOnGround )
+		{
+			ChangeState( ActorStates.Gliding );
+
+			rigidbody.useGravity = false;
+
+			if ( actor.animator )
+			{
+				actor.animator.SetBool( "isJumping", true );
+			}
+		}
+	}
+
+	public void EndGlide()
+	{
+		rigidbody.useGravity = true;
+
+		if ( actor.animator )
+		{
+			actor.animator.SetBool( "isJumping", false );
+		}
+	}
+
+	public void GlideMovement( Vector3 inputVec )
+	{
+		_moveVec = inputVec * _glideHorizontalSpeed;
+		_moveVec.y = -_glideDescentRate;
+
+		rigidbody.velocity = _moveVec;
+	}
+
+	public void RegisterStateMethod( ActorStates state, ActorStateMethod method )
+	{
+		_stateMethodMap.Add( state, method );
+	}
+
+	void ConstrainBumper()
+	{
+		_bumperTransform.position = transform.position;
+		_bumperRigidbody.velocity = rigidbody.velocity;
+	}
+
+	void FollowBumper()
+	{
+		if ( !IsInState( ActorStates.Climbing ) )
+		{
+			Vector3 constrainedPos = transform.position;
+			constrainedPos.x = _bumperTransform.position.x;
+			constrainedPos.z = _bumperTransform.position.z;
+			transform.position = constrainedPos;
+		}
+	}
+
+	void MoveAtSpeed( Vector3 inVec, float moveSpeed )
```

Comment:
It's the direction that you want to move in. I'll rename it to something like `moveDir` since that's more descriptive. As for it being normalized, it shouldn't be because in most cases it's the input value which can scale based on how far the stick is moved over.